internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -876,6 +876,26 @@ func NewClusterRoleBindingForCR(cr *operatorv1beta1.Cryostat) *rbacv1.ClusterRol
 	}
 }
 
+func NewOAuthClusterRoleBindingForCR(cr *operatorv1beta1.Cryostat) *rbacv1.ClusterRoleBinding {
```

Comment:
How could I get more information to debug this error?
Logs from `oc logs -f cryostat-operator-controller-manager-blah`:
```
2021-11-15T19:24:40.410Z	INFO	controllers.Cryostat	Not found	{"Request.Namespace": "", "Name": "cryostat-oauth", "Kind": "*v1.ClusterRoleBinding"}
2021-11-15T19:24:40.415Z	ERROR	controllers.Cryostat	Could not be created	{"Request.Namespace": "", "Name": "cryostat-oauth", "Kind": "*v1.ClusterRoleBinding", "error": "clusterroles.rbac.authorization.k8s.io \"cryostat-oauth\" not found"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
github.com/cryostatio/cryostat-operator/internal/controllers.(*CryostatReconciler).createObjectIfNotExists
	/workspace/internal/controllers/cryostat_controller.go:513
github.com/cryostatio/cryostat-operator/internal/controllers.(*CryostatReconciler).createRBAC
	/workspace/internal/controllers/cryostat_controller.go:570
github.com/cryostatio/cryostat-operator/internal/controllers.(*CryostatReconciler).Reconcile
	/workspace/internal/controllers/cryostat_controller.go:227
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:235
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:198
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.UntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:99
```

Reply:
In this case you're missing the ClusterRole, and Kubernetes is refusing to create the ClusterRoleBinding if the referenced ClusterRole is missing. The existing ClusterRole used by the Cryostat deployment is specified here: https://github.com/cryostatio/cryostat-operator/blob/d7678d83aaf4781d230b693b61fce255b031c808/config/rbac/cryostat_role.yaml.

Since the same ClusterRole can be shared by any number of Cryostat deployments, the operator simply creates it at startup. The same isn't true for the ClusterRoleBinding, which binds that ClusterRole to the unique ServiceAccount belonging to a single Cryostat deployment. That's why the ClusterRole is defined in YAML, and the ClusterRoleBInding is created programmatically.

Feel free to use the existing ClusterRole (and its Binding) for these new permissions you need.

Reply:
If I try to add permissions to the ClusterRole in `cryostat_role.yaml`, it looks like the `cryostat-operator-service-account` that creates this ClusterRole needs the same permissions before it can grant them to the ClusterRole. Where are the service account permissions set?
```
ERROR	controller-runtime.manager.controller.cryostat	
Reconciler error	{"reconciler group": "operator.cryostat.io", "reconciler kind": "Cryostat",
 "name": "cryostat-sample", "namespace": "default", "error":
 "clusterrolebindings.rbac.authorization.k8s.io \"cryostat-d6f4ddc1e7a1dbb5e1a1b9cac4fa216d0d90e8f734fe70ebb132a5c400f1c345\" is forbidden: 
user \"system:serviceaccount:default:cryostat-operator-service-account\" (groups=
[\"system:serviceaccounts\" \"system:serviceaccounts:default\" \"system:authenticated\"]) is
 attempting to grant RBAC permissions not currently held:\n{APIGroups: 
<list of new permissions>}
```

Reply:
The Role and ClusterRole for the operator deployment are generated using marker comments, which get processed at build time by the `make manifests` target [1]. Here's where you can add the new permission(s):
https://github.com/cryostatio/cryostat-operator/blob/d7678d83aaf4781d230b693b61fce255b031c808/internal/controllers/cryostat_controller.go#L104-L120

To add permissions to the ClusterRole, don't specify the namespace attribute, otherwise they'll get added to the namespaced Role instead.

[1] https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/#specify-permissions-and-generate-rbac-manifests